viewing_party/party.py

```diff
@@ -1,23 +1,145 @@
+        genre = movie["genre"]
+        if genre in genre_count:
+            genre_count[genre] += 1
+        else:
+            genre_count[genre] = 1
+    return max(genre_count, key=lambda genre: genre_count[genre])
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_unique_watched(user_data):
+
+    user_movies = user_data["watched"]
+    friend_movies = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friend_movies.append(movie)
+
+    unique_movies = []
+
+    for user_movie in user_movies:
+        if user_movie not in friend_movies and user_movie not in unique_movies:
+            unique_movies.append(user_movie)
+
+    print(f"{unique_movies = }")
+    return unique_movies
+
+    # user_movies = set([movie['title'] for movie in user_data['watched']])
+    # friends_movies = set([movie['title'] for friend in user_data['friends'] for movie in friend['watched']])
+    # unique_movies = user_movies.difference(friends_movies)
+    # unique_movies = [{'title': movie} for movie in unique_movies]
+    # print(f"{unique_movies = }")
+    # return unique_movies
+
+def get_friends_unique_watched(user_data):
+    user_movies = user_data["watched"]
+    friend_movies = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friend_movies.append(movie)
+            
+    unique_movies = []
+
+    for friend_movie in friend_movies:
+        if friend_movie not in user_movies and friend_movie not in unique_movies:
+            unique_movies.append(friend_movie)
+
+    # print(f"{unique_movies = }")
+    return unique_movies
+
+#     user_movies = set([movie['title'] for movie in user_data['watched']])
+#     friends_movies = set([movie['title'] for friend in user_data['friends'] for movie in friend['watched']])
+#     unique_movies = friends_movies.difference(user_movies)
+#     return [{'title': movie} for movie in unique_movies]
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    movies_not_watched = get_friends_unique_watched(user_data)
+    recommendations = []
+    sub_list = user_data["subscriptions"]
+    for movie in movies_not_watched:
+        if movie["host"] in sub_list:
+                recommendations.append(movie)
+    return recommendations
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+
+    recommendations = []
+    not_watched_yet = get_friends_unique_watched(user_data)
+    fav_genre = get_most_watched_genre(user_data)
+
+    for movie in not_watched_yet:
+        if movie["genre"] == fav_genre:
+            recommendations.append(movie)
+    return recommendations

```

Comment:
Really well done with this function too! I love the way you made each part its own call to a function. And once again, your instinct to use helper functions is great!